.github/workflows/release.yml

```diff
@@ -0,0 +1,267 @@
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.m2/repository
+          !~/.m2/repository/org/projectnessie
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    # TODO Maybe better to _not_ use cached dependencies just to be sure that everything's available at time of release --> remove this one?
+    - name: Cache local NPM repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    # TODO Maybe better to _not_ use cached dependencies just to be sure that everything's available at time of release --> remove this one?
+    - name: Cache local Gradle cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+        restore-keys: ${{ runner.os }}-gradle
+
+    - name: Prepare git config
+      run: |
+        git config --global user.email "nessie-release-workflow-noreply@projectnessie.org"
+        git config --global user.name "Projectnessie Release Workflow [bot]"
+
+        # Update the remote repo URL to include the secret so this job can push to the repo
+        UPSTREAM="$(git remote)"
+        REPO_OWNER="$(echo ${GITHUB_REPOSITORY} | cut -d/ -f2)"
+        git remote set-url ${UPSTREAM} https://${REPO_OWNER}:${{ secrets.GITHUB_TOKEN }}@github.com/${GITHUB_REPOSITORY}
+
+    - name: Install Python dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r python/requirements_lint.txt
+        pip install tox-gh-actions bumpversion twine
+
+    - name: Bump release version ${{ github.event.inputs.releaseVersion }}
+      env:
+        RELEASE_VERSION: ${{ github.event.inputs.releaseVersion }}
+        GIT_TAG: nessie-${{ github.event.inputs.releaseVersion }}
+      run: |
+        ./mvnw versions:set -DgenerateBackupPoms=false -DnewVersion=${RELEASE_VERSION}
+        (cd python ; bumpversion --no-commit --new-version ${RELEASE_VERSION} minor)
+
+    - name: Build via Maven and run tests
+      env:
+        MAVEN_GPG_PASSPHRASE: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
+        SPARK_LOCAL_IP: localhost
+      run: |
+        ./mvnw install -Pnative -Pjdk8-tests -Pcode-coverage -Prelease
+
+    - name: Tag the Docker image
+      run: |
+        # We can always use projectnessie/nessie here because it's a release
+        docker tag projectnessie/nessie:${RELEASE_VERSION} projectnessie/nessie:latest
+
+    - name: Build Gradle plugin and run tests
+      run: |
+        (cd tools/apprunner-gradle-plugin/ ; ./gradlew build)
+
+    - name: Build pynessie and run tests
+      run: |
+        cd python
+        tox
+        python setup.py sdist bdist_wheel
+
+    # TODO Remove this one - b/c it's codecov?
```

Comment:
agreed

Reply:
I found another way to hide secrets from codecov :)